src/cache.rs

```diff
@@ -160,22 +160,49 @@ impl Cache {
         let platform = self.get_platform_dir();
 
         // Search for the page in the platform specific directory
-        if let Some(pf) = platform {
-            let path = platforms_dir.join(&pf).join(&page_filename);
-            if path.exists() && path.is_file() {
-                return Some(path);
+        let pf_path = if let Some(pf) = platform {
+            let p = platforms_dir.join(&pf).join(&page_filename);
+            if p.exists() && p.is_file() {
+                Some(p)
+            } else {
+                None
             }
-        }
+        } else {
+            None
+        };
 
         // If platform is not supported or if platform specific page does not exist,
         // look up the page in the "common" directory.
-        let path = platforms_dir.join("common").join(&page_filename);
+        let common_path = platforms_dir.join("common").join(&page_filename);
+        let common_path = if common_path.exists() && common_path.is_file() {
+            Some(common_path)
+        } else {
+            None
+        };
+
+        // Get custom pages directory
+        let custom_env = if let Ok(value) = env::var("TEALDEER_CUSTOM_PAGES_DIR") {
+            value
+        } else {
+            String::from("../pages.custom")
+        };
 
-        // Return it if it exists, otherwise give up and return `None`
-        if path.exists() && path.is_file() {
-            Some(path)
+        // Search the custom directory for matching files
+        let custom_path = platforms_dir.join(custom_env).join(&page_filename);
+        let custom_path = if custom_path.exists() && custom_path.is_file() {
+            Some(custom_path)
         } else {
             None
+        };
+
+        // Return pages if they exists, otherwise give up and return `None`
+        match (pf_path, common_path, custom_path) {
+            (Some(pfp), _, Some(cup)) => Some(vec![pfp, cup]),
+            (Some(pfp), _, None) => Some(vec![pfp]),
+            (None, Some(cop), Some(cup)) => Some(vec![cop, cup]),
+            (None, Some(cop), None) => Some(vec![cop]),
+            (None, None, Some(cup)) => Some(vec![cup]),
+            (None, None, None) => None,
```

Comment:
The way this is implemented right now means that if a platform specific page exists, instead of a single path lookup we now need to do three (because the early return was removed). Since I/O can be slow, this should be avoided if possible. For example, if a platform-specific page exists, the common directory lookup doesn't need to be done.

Your current approach always appends the custom pages to the existing pages. However, in some cases people might want to replace entire pages, or add a page if it does not exist yet. So here's an idea:

- If a page filename ends with `.page`, it should replace the entire page, no matter whether it already exists or not
- If a page filename ends with `.patch`, it should be appended to an existing page, if it exists.

The logic could be implemented as follows:

1. Look up custom page (`<name>.page`). If it exists, return it directly,
2. Look up custom patch (`<name>.patch`). If it exists, store it in a variable.
3. Look up platform-specific page. If it exists, return page plus the patch above (if set).
4. Look up page from the "common" directory. If it exists, return page plus the patch above (if set).
5. As a fallback, return an empty vector.

What do you think?

Reply:
Sounds great I'll make those changes after Thanksgiving, thanks for the feedback!

Reply:
Changes made, let me know what you think 😄